network/ocsp.go

```diff
@@ -172,7 +173,7 @@ func (c DefaultOCSPClient) Query(commonName string, clientCert, issuerCert *x509
 	if err != nil {
 		return nil, err
 	}
-	ocspResponse, err := ocsp.ParseResponse(output, issuerCert)
+	ocspResponse, err := ocsp.ParseResponse(output, nil)
```

Comment:
why nil?

Reply:
Actually, this was a mistake. First, I thought there should be `issuer` certificate, as default OCSP responder certificate (if the responder didn't send own one attached to response). Then I misunderstood some things and put `nil` here.
What we have now: `ocsp.ParseResponse(output, issuerCert)`. So it [tries](https://github.com/golang/crypto/blob/c1f2f97bffc9/ocsp/ocsp.go#L550) to use attached cert if exists, otherwise tries passed one (`issuerCert`) who signed the cert we ask about.
Inside `ParseResponse()` there is a check that if `issuer != nil` then attached OCSP responder cert should be signed by `issuer`. This way, we only accept OCSP responses signed by either issuer itself or some intermediate cert signed by this exact issuer.